cli/preprocessor.go

```diff
@@ -96,15 +96,15 @@ func analyze(
 		*compiler = "cpp"
 	}
 
-	// Ensure value of comment char
-	if *lineCommentIden == "" && *blockCommentIdenOpen == "" && *blockCommentIdenClose == "" {
-		*lineCommentIden, *blockCommentIdenOpen, *blockCommentIdenClose = getCommentIdenFromLang(lang, *fileInput)
-	} else if (*blockCommentIdenOpen == "" && *blockCommentIdenClose != "") || (*blockCommentIdenOpen != "" && *blockCommentIdenClose == "") {
-		util.Error("You cannot specify only one of blockCommentIdenOpen and blockCommentIdenClose. Please specify both or none.", true)
-	}
-
 	// Get raw data strings
 	if !modeSingle {
+		// Ensure value of comment char
+		if *lineCommentIden == "" && *blockCommentIdenOpen == "" && *blockCommentIdenClose == "" {
+			*lineCommentIden, *blockCommentIdenOpen, *blockCommentIdenClose = getCommentIdenFromLang(lang, *fileInput)
+		} else if (*blockCommentIdenOpen == "" && *blockCommentIdenClose != "") || (*blockCommentIdenOpen != "" && *blockCommentIdenClose == "") {
+			util.Error("You cannot specify only one of blockCommentIdenOpen and blockCommentIdenClose. Please specify both or none.", true)
+		}
+
```

Comment:
I'm wondering how this fixes #50 since you wrapped the if statement around another error and not 

```
Please use lang 'auto' only in combination of valid file paths as file input
```

Reply:
@Splines the if statement is also wrapped around the method call to `getCommentIdenFromLang()` where this error message is thrown, so when the method isn't even called, no error can be thrown ;)

Reply:
ah, I see 👌